### PR TITLE
ci: setup pipeline

### LINKED ISSUE #4: ci: introduce CI using Github Actions
Need to create CI pipeline to do
- static code check
- run unit tests
- run integration tests

### PR DESCRIPTION
**Pull request by askalt**
Closes #4

### CODE REVIEW

File: key_bind_func.go [hunk 1]
```diff
@@ -19,7 +19,8 @@ func DeleteChar(buf *Buffer) {
 
 // DeleteWord Delete word before the cursor
 func DeleteWord(buf *Buffer) {
-	buf.DeleteBeforeCursor(len([]rune(buf.Document().TextBeforeCursor())) - buf.Document().FindStartOfPreviousWordWithSpace())
+	buf.DeleteBeforeCursor(len([]rune(buf.Document().TextBeforeCursor())) - buf.Document().
```
Comment (psergee) [inline key_bind_func.go:22]: Let's break on `-`.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+name: Tests
+
+on:
+  push:
+    branches-ignore:
+      - 'master'
```
Comment (psergee) [inline .github/workflows/tests.yml:7]: Tests are required on push to `master` also.

File: completion_test.go [hunk 3]
```diff
@@ -112,6 +171,7 @@ func TestFormatShortSuggestion(t *testing.T) {
 			t.Errorf("[scenario %d] Want %d but got %d\n", i, s.exWidth, width)
 		}
 		if !reflect.DeepEqual(actual, s.expected) {
+			fmt.Println(actual)
```
[the remark this answers is not in the document]
Comment (askalt) [reply completion_test.go]: Fixed.

File: filter.go [hunk 4]
```diff
@@ -23,7 +23,8 @@ func FilterContains(completions []Suggest, sub string, ignoreCase bool) []Sugges
 // FilterFuzzy checks whether the completion.Text fuzzy matches sub.
 // Fuzzy searching for "dog" is equivalent to "*d*o*g*". This search term
 // would match, for example, "Good food is gone"
-//                               ^  ^      ^
+//
+//	^  ^      ^
```
[the remark this answers is not in the document]
Comment (askalt) [reply filter.go]: Fixed.

File: filter_test.go [hunk 5]
```diff
@@ -133,7 +133,8 @@ func TestFilter(t *testing.T) {
 	}
 
 	for _, s := range scenarioTable {
-		if actual := s.filter(s.list, s.substr, s.ignoreCase); !reflect.DeepEqual(actual, s.expected) {
+		if actual := s.filter(s.list, s.substr, s.ignoreCase); !reflect.DeepEqual(actual, s.
```
[the remark this answers is not in the document]
Comment (askalt) [reply filter_test.go]: Fixed.

File: key_bind_func.go (hunk 1 above)
Comment (askalt) [reply key_bind_func.go]: Fixed.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (askalt) [reply .github/workflows/tests.yml]: Fixed.

File: .github/actions/static-code-check/action.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+name: "Run static code check"
+description: "Performs static code checks."
+
+runs:
+  using: "composite"
+  steps:
+      - name: Install tests requirements
+        run: |
+          pip3 install -r test/requirements.txt
```
[the remark this answers is not in the document]
Comment (askalt) [reply .github/actions/static-code-check/action.yml:15]: Venv added.

**Review by psergee — APPROVED** [review]

**Pull request merged**